backend/src/modules/Boards/service/validateAddingAnnouncement.js

```diff
@@ -0,0 +1,22 @@
+import joi from 'joi';
+import joiObjectId from 'joi-objectid';
+const myJoiObjectId = joiObjectId(joi);
+
+export const validateAddingAnnouncement = (params, repositoryType) => {
+  let addingNewAnnouncementValidationSchema;
+  if (repositoryType === 'MongoBoardsRepository') {
+    addingNewAnnouncementValidationSchema = joi.object({
+      boardId: myJoiObjectId().required(),
+      announcementId: myJoiObjectId().required(),
+    });
+  } else {
+    addingNewAnnouncementValidationSchema = joi.object({
+      boardId: joi.string().required(),
+      announcementId: myJoiObjectId().required(),
+    });
+  }
+
+  return addingNewAnnouncementValidationSchema.validate(params);
+};
+
+export default validateAddingAnnouncement;
```

Comment:
Nie rozumiem totalnie tej walidacji xD

Reply:
Mongo tworzy nam ID jako ObjectID. 
W in memory mamy uuid.

Reply:
Więc róznica jest między linią 9 i 14.